Lib/fontTools/cffLib.py

```diff
@@ -93,21 +127,35 @@ def toXML(self, xmlWriter, progress=None):
 		xmlWriter.endtag("GlobalSubrs")
 		xmlWriter.newline()
 
-	def fromXML(self, name, attrs, content):
+	def fromXML(self, name, attrs, content, otFont):
+		self.otFont = otFont
+
+		# set defaults. These will be replaced if there are entries for them in the XML file.
 		if not hasattr(self, "GlobalSubrs"):
 			self.GlobalSubrs = GlobalSubrsIndex()
+		if not hasattr(self, "major"):
 			self.major = 1
+		if not hasattr(self, "minor"):
 			self.minor = 0
+		if not hasattr(self, "hdrSize"):
 			self.hdrSize = 4
-			self.offSize = 4  # XXX ??
+		if not hasattr(self, "offSize"):
+			self.offSize = 4 # this will be recalculated when the cff is compiled.
+			
 		if name == "CFFFont":
-			if not hasattr(self, "fontNames"):
-				self.fontNames = []
-				self.topDictIndex = TopDictIndex()
-			fontName = attrs["name"]
-			topDict = TopDict(GlobalSubrs=self.GlobalSubrs)
-			topDict.charset = None  # gets filled in later
-			self.fontNames.append(fontName)
+			isCFF2 = (self.major == 2)
+			if self.major == 1:
```

Comment:
Maybe remove isCFF2 and always just check self.major directly?  It's very inconsistent right now.